Sources/SWBCore/DependencyResolution.swift

```diff
@@ -607,27 +627,75 @@ extension SpecializationParameters {
         return configuredTargets
     }
 
+    /// Determines whether a top level target should be configured for the given platform.
     nonisolated func isTargetSuitableForPlatform(_ target: Target, parameters: BuildParameters, imposedParameters: SpecializationParameters?) -> Bool {
+        guard let runDestination = parameters.activeRunDestination else { return false }
+
         // Filter any overrides from the parameters, because they can shadow any original "auto" values.
         let filteredParameters = parameters.withoutOverrides
-        var settings = buildRequestContext.getCachedSettings(filteredParameters, target: target)
+        let settings = buildRequestContext.getCachedSettings(filteredParameters, target: target)
+        let supportedPlatforms = settings.globalScope.evaluate(BuiltinMacros.SUPPORTED_PLATFORMS)
+
+        // If the target is being specialized, all that matters is that it supports the imposed parameters. Note that
+        // there's no point actually imposing the specialization to check against the parameters - they will
+        // necessarily match after they are imposed.
         if settings.enableTargetPlatformSpecialization, let imposedParameters {
-            // This is for targets that have SDKROOT set to 'auto' (like Swift Packages) and get settings "imposed" on them.
-            settings = buildRequestContext.getCachedSettings(imposedParameters.imposed(on: filteredParameters, workspaceContext: workspaceContext), target: target)
+            return imposedParameters.supports(platforms: supportedPlatforms)
```

Comment:
I think this is technically reducing the specializations that are actually possible in the non-index case. While supported platforms will change the behavior, we will always impose a platform (see https://github.com/swiftlang/swift-build/blob/main/Sources/SWBCore/DependencyResolution.swift#L764) since "auto" isn't a valid one.

Reply:
Yeah, just to be clear, my expectation is that this doesn't change the current behavior in any way. I'm happy to do whatever you want here, I included this mainly because I had already refactored it to help myself understand it better. But I can pull out and put up another PR with it + remove the check entirely if you'd prefer.

Reply:
It was more an observation since we were recently talking about how the index build differs from the real one and this is a difference that I hadn't been aware of.

Reply:
Spoke offline, but to summarize things:
1. Only the index path uses this check (I'll rename the function)
2. The general idea here is that since we're configuring for all platforms, there's no reason to specialize for an unsupported platform - we'll separately configure for that platform anyway
3. I'll change this a bit regardless. It's a little confusing because it's only correct under the assumption that the active run destination matches the imposed parameters (which they do, because that's how we build them). But also it means we can skip the bottom `supportedPlatforms` check entirely, since the `targetPlatform.name` check is sufficient outside of the specialization case